Standalone binary

weierophinney opened the pull request:
In seeing some people attempt to use the library, I realized it may be better to ship it as a standalone binary:

- The class it generates has no dependencies.
- Generally, you will generate the preload file once, and then manipulate it to provide additional rules. Once generated, you no longer need the tooling

This patch provides a `bin/phly-opcache-preload` file exposed as a vendor binary in order to allow global installation. When used in this way, the commands become `generate:preload-file` and `generate:ini`.

weierophinney committed:
feat: standalone binary

In seeing some people attempt to use the library, I realized it may be
better to ship it as a standalone binary:

- The class it generates has no dependencies.
- Generally, you will generate the preload file once, and then
  manipulate it to provide additional rules. Once generated, you no
  longer need the tooling

This patch provides a `bin/phly-opcache-preload` file exposed as a
vendor binary in order to allow global installation. When used in this
way, the commands become `generate:preload-file` and `generate:ini`.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

michalbundyra commented on composer.json:
```diff
@@ -14,7 +14,8 @@
     "require": {
         "php": "~7.4.4",
-        "laminas/laminas-cli": "dev-master as 0.1.0"
+        "laminas/laminas-cli": "dev-master as 0.1.0",
```
I think we should drop laminas-cli dependency now.

michalbundyra commented on composer.json:
```diff
@@ -14,7 +14,8 @@
         "php": "~7.4.4",
-        "laminas/laminas-cli": "dev-master as 0.1.0"
+        "laminas/laminas-cli": "dev-master as 0.1.0",
+        "symfony/console": "^5.0"
```
I think we should allow also previous version of symfony/console (maybe the same as we have in laminas-cli symfony/console 4.1 || 5.0)

michalbundyra reviewed (changes requested):
@weierophinney please see my comments below, thanks!

weierophinney committed:
fix: do not expose as laminas-cli commands

- Simplifies detailing usage.
- Promotes idea of standalone usage.
- Does not require exposing as a component or module.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney commented:
@michalbundyra Noted your points, and pushed a change that adopts your feedback, as well as:

- Removes the `ConfigProvider` and `Module` classes, as they are then no longer relevant.
- Removes verbiage in the `README.md` about usage with laminas-cli.

michalbundyra commented:
@weierophinney I kinda like that it was available in laminas-cli when laminas-cli is installed, but the problem is that then the module/config-provider must be registered, and it is not something we want to have in the applications. So the changes, you just have made, make sense.

michalbundyra reviewed (approved):
🚢

weierophinney commented:
> I kinda like that it was available in laminas-cli when laminas-cli is installed, but the problem is that then the module/config-provider must be registered, and it is not something we want to have in the applications.

Exactly. It's really meant to be used once in the application, no more.

Thanks for your feedback!
